Sources/SwiftCuckoo/Lap.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+
+/// A model representing an individual lap within a session.
+///
+/// The `Lap` structure tracks the start time, end time, and calculates the duration
+/// of a single lap, providing a way to monitor intervals within a task or session.
+public struct Lap: Equatable, Hashable {
+
+    /// Defines the different states of a  `Lap`.
+    public enum LapStatus: Equatable {
+        case active                /// Indicates the lap is active.
+        case inactive              /// Indicates the lap is inactive.
+    }
+
+    /// A unique identifier for the lap instance.
+    public var id: Identifier
+
+    /// The start time of the lap.
+    public var startTime: Date?
+
+    /// The optional end time of the lap. Defaults to `nil` if the lap is ongoing.
+    public var endTime: Date?
+
+    /// Initializes a new lap instance with a specified start time and an optional end time.
+    ///
+    /// - Parameters:
+    ///   - startTime: The start time of the lap.
+    ///   - endTime: The end time of the lap. Defaults to `nil` for ongoing laps.
+    public init(id: Identifier, startTime: Date? = nil, endTime: Date? = nil) {
+        self.id = id
+        self.startTime = startTime
+        self.endTime = endTime
+    }
```

Comment:
If we decide to remove the identifier, we can replace it with UUID.

For example:

```suggestion
    public init(startTime: Date? = nil, endTime: Date? = nil) {
        self.id = UUID()
        self.startTime = startTime
        self.endTime = endTime
    }
```

Reply:
I like this route 👍